integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -495,6 +511,65 @@ def post(self, request, *args, **kwargs):
             },
         )
 
+    # pylint: disable=consider-using-assignment-expr
+    @staticmethod
+    def cleanup_autosaves(objects, language, keep=3):
+        """
+        Delete all but the last 3 AUTO_SAVEs
+        and renumber all affected versions
+        to be continuous.
+        """
+        all_versions = objects.filter(language=language)
+        auto_saves = all_versions.filter(status=status.AUTO_SAVE)
+        versions = list(reversed(list(map(lambda x: x.version, auto_saves))))
+        if not versions:
+            # nothing to clean up
+            return 0
+        earliest = versions[0]
+        affected_versions = all_versions.filter(version__gte=earliest)
+        try:
+            lastN = versions[-keep]
+            to_delete = auto_saves.filter(version__lt=lastN)
+            v_deleted = list(reversed(list(map(lambda x: x.version, to_delete))))
+            print(f"##### v_deleted = {v_deleted}")
```

Comment:
We left the print statement in because we asked for careful testing around the `bulk_update()` call and fear `IntegrityError`s, in which case our debug prints might help identify the exact situation. We planned on deleting them when no issue would be found, before merging.

*On a side note, this is where I'm still a bit confused about our approach. Sure, discussions on commits get more or less lost as soon as new ones are pushed, but using PRs to get in part preliminary feedback has the impression that everything is final while I'd not even look at ones marked as draft because everything can still change anyway, or so is my gut reaction. This is a discussion for a later time, maybe the next conference*

Reply:
I think the only way around this problem is that you explicitly state in the PR description which parts you want to change anyway and don't need to be mentioned in the review...

So I guess there are two modes of reviews: the complete review (which just mentions all problems it finds in a PR) and the preliminary review (which has to be either limited to specific questions in the PR descriptions: "please only look at x", or specific points have to be excluded: "please don't look at y")...